code/game/turfs/turf.dm

```diff
@@ -132,6 +132,31 @@
 	stack_trace("Non movable passed to turf CanPass : [mover]")
 	return FALSE
 
+//returns which obstruction would prevent entering. Update if /turf/enter is updated.
+/turf/proc/CheckEnteringObstruction(atom/movable/mover, atom/forget)
```

Comment:
this is really unacceptable, to have one proc that mirrors the work another proc does.

you should refactor them to share logic